[java] Add JSpecify nullable annotations to exception classes

iampopovich opened the pull request:
### **User description**
<!-- Thanks for Contributing to Selenium! -->
<!-- Please read our contribution guidelines: https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md -->

### 🔗 Related Issues
partially fixes #14291

### 💥 What does this PR do?
same as 
#16024
#16025

### 🔧 Implementation Notes
This pull request introduces nullness annotations from the `jspecify` library to improve type safety in exception classes within the Selenium Java codebase. The changes ensure that method parameters and classes explicitly handle nullable values, reducing ambiguity and potential runtime errors.

### Addition of `jspecify` nullness annotations:

* [`java/src/org/openqa/selenium/InvalidCookieDomainException.java`](diffhunk://#diff-ab89d3c82f8236d8c79143a6d53ad44ea24479adbd34733201c1864d39e7dda6R20-R40): Added `@NullMarked` to the class and `@Nullable` annotations to method parameters for constructors.
* [`java/src/org/openqa/selenium/UnableToSetCookieException.java`](diffhunk://#diff-b9dcc50f8e49db5fbdf70c65e0058566e9c382129de602acd497330dd80ad1a1R20-R40): Applied `@NullMarked` to the class and `@Nullable` to constructor parameters.
* [`java/src/org/openqa/selenium/devtools/DevToolsException.java`](diffhunk://#diff-c58c5f9081d9e62cb230c086759a1e5b06c713f5635309d60576c1c50d215291R20-R35): Introduced `@NullMarked` for the class and `@Nullable` for constructor parameters, ensuring proper handling of nullable values.
* [`java/src/org/openqa/selenium/json/JsonException.java`](diffhunk://#diff-4a16c7edfee9b2818b3b2ae7445b1388bce74327fc45443b8b039432edcecf76R20-R35): Updated the class with `@NullMarked` and added `@Nullable` annotations to constructor parameters.
* [`java/src/org/openqa/selenium/remote/ScreenshotException.java`](diffhunk://#diff-c93bd5edeed5cfa779a4238e9b35f7b746dd9203a4b400186eef936042ee41b8R20-R35): Enhanced the class with `@NullMarked` and applied `@Nullable` annotations to constructor parameters.

### 💡 Additional Considerations
<!--- Are there any decisions that need to be made before accepting this PR? -->
<!--- Is there any follow-on work that needs to be done? (e.g., docs, tests, etc.) -->

### 🔄 Types of changes
<!-- ✂️ Please delete anything that doesn't apply -->
- Cleanup (formatting, renaming)


___

### **PR Type**
Enhancement


___

### **Description**
- Add JSpecify nullable annotations to exception classes

- Improve type safety with @NullMarked and @Nullable

- Update build dependencies for jspecify library


___

### **Changes diagram**

```mermaid
flowchart LR
  A["Exception Classes"] --> B["Add @NullMarked"]
  A --> C["Add @Nullable to parameters"]
  B --> D["Type Safety Improvement"]
  C --> D
  E["Build Files"] --> F["Add jspecify dependency"]
```


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>5 files</summary><table>
<tr>
  <td><strong>InvalidCookieDomainException.java</strong><dd><code>Add JSpecify annotations to cookie exception</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16026/files#diff-ab89d3c82f8236d8c79143a6d53ad44ea24479adbd34733201c1864d39e7dda6">+7/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>UnableToSetCookieException.java</strong><dd><code>Add JSpecify annotations to cookie exception</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16026/files#diff-b9dcc50f8e49db5fbdf70c65e0058566e9c382129de602acd497330dd80ad1a1">+7/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>DevToolsException.java</strong><dd><code>Add JSpecify annotations to DevTools exception</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16026/files#diff-c58c5f9081d9e62cb230c086759a1e5b06c713f5635309d60576c1c50d215291">+6/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>JsonException.java</strong><dd><code>Add JSpecify annotations to JSON exception</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16026/files#diff-4a16c7edfee9b2818b3b2ae7445b1388bce74327fc45443b8b039432edcecf76">+6/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ScreenshotException.java</strong><dd><code>Add JSpecify annotations to screenshot exception</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16026/files#diff-c93bd5edeed5cfa779a4238e9b35f7b746dd9203a4b400186eef936042ee41b8">+6/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>BUILD.bazel</strong><dd><code>Add jspecify dependency to devtools build</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16026/files#diff-076b9e9543c85d763a79b700d8bcb2492886b6753c8c44552672583902acfe05">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>BUILD.bazel</strong><dd><code>Add jspecify dependency to json build</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16026/files#diff-e8edc082e1858956cde859f3bf0afbb1690ad41089125241704faf72b02878a6">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

iampopovich committed:
Add JSpecify nullable annotations to exception classes

iampopovich committed:
fix dependencies on jspecify

iampopovich committed:
add dependency jspecify to build bazel

iampopovich committed:
Merge branch 'trunk' into feat-14291/add-jspecify-annotations-to-exception-classes-pt3

iampopovich commented:
@pujagani @mk868 please review

iampopovich committed:
Merge branch 'trunk' into feat-14291/add-jspecify-annotations-to-exception-classes-pt3

mk868 commented on java/src/org/openqa/selenium/json/BUILD.bazel:
```diff
@@ -14,5 +14,6 @@ java_export(
     ],
     deps = [
         "//java/src/org/openqa/selenium:core",
+        "@maven//:org_jspecify_jspecify",
```
I think, it's better to use `artifact("org.jspecify:jspecify"),`
At the top, an "artifact" load will be required:
```
load("//java:defs.bzl", "artifact", "java_export")
```

mk868 replied:
Maybe I'm wrong,
Were there problems with `artifact("org.jspecify:jspecify"),` here?

iampopovich replied:
i saw references in other BUILD files and missed usage of artifact() function. thanks !

iampopovich replied:
will be fixed in #16635

iampopovich committed:
Merge branch 'trunk' into feat-14291/add-jspecify-annotations-to-exception-classes-pt3

diemol committed:
Merge branch 'trunk' into feat-14291/add-jspecify-annotations-to-exception-classes-pt3

asolntsev reviewed (commented):
@iampopovich Why the exception parameters are nullables?

I think it's better to make them non-nullable. It doesn't make sense to throw e.g. `new ScreenshotException(null)`. 

I've checked Selenium code - there is NO occurrences of
* `new ScreenshotException(null`
* `new JsonException(null`
* `new DevToolsException(null`
* `new UnableToSetCookieException(null`
* `new InvalidCookieDomainException(null`

iampopovich commented:
@asolntsev 
InvalidCookieDomainException presented in .NET client with nullable parameters , i've added the same for the java client
https://github.com/SeleniumHQ/selenium/blob/f7cc7cb6b4834d8b0c3a0cfcf945484d3491716c/dotnet/src/webdriver/InvalidCookieDomainException.cs#L43 
same for UnableToSetCookieException here 
https://github.com/SeleniumHQ/selenium/blob/f7cc7cb6b4834d8b0c3a0cfcf945484d3491716c/dotnet/src/webdriver/UnableToSetCookieException.cs#L43
same for InvalidCookieDomainException here
https://github.com/SeleniumHQ/selenium/blob/f7cc7cb6b4834d8b0c3a0cfcf945484d3491716c/dotnet/src/webdriver/InvalidCookieDomainException.cs#L43


For the remaining exceptions, I added nullable parameters similar to the existing ones.
